File: java/Person.java
public class Person {
    int ID;
    String name;

    public Person(int ID, String name) {
        this.ID = ID;
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Person person = (Person) o;
        return ID == person.ID && name.equals(person.name);
    }

    @Override
    public int hashCode() {
        return ID * 31 + name.hashCode();
    }
}

File: py/person.py
class Person:
    def __init__(self, ID: int, name: str):
        self.ID = ID
        self.name = name
    def __eq__(self, other):
        return isinstance(other, Person) and self.ID == other.ID and self.name == other.name
    def __hash__(self):
        return hash((self.ID, self.name))
